### PR TITLE
Feat/sns my votes

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

Display user votes on proposal detail page.

# Changes

- refactor `CompactNeuronInfo` to use `idString` instead of `NeuronId`
- refactor `<MyVotes>` to get `CompactNeuronInfo` in props
- use `<MyVotes>` in `<SnsCard>`
- util `votedSnsNeurons`
- util `votedSnsNeuronDetails `
- util `votedSnsNeuronDetails `

# Tests

- new utils 
- display my votes in SnsVotingCard

# Screenshot
## SNS
image

# NNS (after refactoring)
image

### CODE REVIEW

File: frontend/src/tests/lib/components/proposal-detail/VotingCard/SnsVotingCard.spec.ts [hunk 1]
```diff
@@ -140,6 +149,28 @@ describe("SnsVotingCard", () => {
     expect(container.querySelector("button[disabled]")).toBeNull();
   });
 
+  it.only("should display my votes", async () => {
```
Comment (lmuntaner) [inline frontend/src/tests/lib/components/proposal-detail/VotingCard/SnsVotingCard.spec.ts:152]: Remove the `only`?

File: frontend/src/tests/lib/utils/sns-neuron.utils.spec.ts [hunk 2]
```diff
@@ -1762,4 +1765,193 @@ describe("sns-neuron utils", () => {
...
     });
   });
+
+  describe("getSnsNeuronVote", () => {
+    const testProposal: SnsProposalData = {
+      ...mockSnsProposal,
+      proposal_creation_timestamp_seconds: 100n,
+      ballots: [
+        [
+          "010203",
```
Comment (lmuntaner) [inline frontend/src/tests/lib/utils/sns-neuron.utils.spec.ts:1775]: Is this the hex string representation?

**Review by lmuntaner — APPROVED** [review]
Looking very good!

I added a couple of comments. I'm surprised about them, but I want to confirm it.

Already approving though.

File: frontend/src/tests/lib/utils/sns-neuron.utils.spec.ts (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/utils/sns-neuron.utils.spec.ts:1775]: Exactly.

File: frontend/src/tests/lib/components/proposal-detail/VotingCard/SnsVotingCard.spec.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/components/proposal-detail/VotingCard/SnsVotingCard.spec.ts:152]: Good catch! Thanks, removed.

File: frontend/src/lib/utils/sns-neuron.utils.ts [hunk 3]
```diff
@@ -728,3 +728,60 @@ export const votableSnsNeurons = ({
...
+
+/** Returns the neurons that have voted on the proposal (based on proposal ballots) */
+export const votedSnsNeurons = ({
+  neurons,
+  proposal,
+}: {
+  neurons: SnsNeuron[];
+  proposal: SnsProposalData;
+}): SnsNeuron[] => {
+  const votedNeuronIds = new Set(
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/lib/utils/sns-neuron.utils.ts:740]: Yes, this is the structure `[[neuronIdAsString, {...}], ...]`

**Pull request merged**